### ISSUE TITLE
Parse error when using short arrow functions as default value in destructured object function argument. 

### ISSUE THREAD

**Issue by WRidder** [eslint/espree#219]
I have the following construct:

```javascript
// Function accepts an object with an action property. 
// If the property is not set, set a default arrow function as value
// If no object is passed at all, use an empty object as initial value as function argument.
const someVar = ({actions = () => ({})} = {}) => {

};
```

This raises an error when running eslint: `parsing error: unexpected token =>`

However, the following does not raise an error:
```javascript
const defaultFunction = () => ({});
const someVar = ({actions = defaultFunction)} = {}) => {

};
```

Both versions do work correctly when executed btw.

My .eslintrc config:
```javascript
{
  "extends": "airbnb",
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "globals": {
    "React": true
  },
  "rules": {
    "valid-jsdoc": 1,
    "quotes": [
      2,
      "double"
    ],
    "new-cap": 0,
    "brace-style": [2, "stroustrup", { "allowSingleLine": false }],
    "id-length": [2, {"min": 1, "properties": "never"}],
    "no-param-reassign": 0,

    // Disable until Flow supports let and const
    "no-var": 0,
    "vars-on-top": 0,

    // Disable comma-dangle unless need to support it
    "comma-dangle": 0
  },
  "root": true
}
```

Running on eslint versions:
```
    "eslint": "1.9.0",
    "eslint-config-airbnb": "1.0.0",
    "eslint-plugin-react": "3.9.0",
```
`

**Comment by gyandeeps** [comment]
This should be fixed with `3.0.0` espree. It not yet released but the changes are in master now.
We do have a `3.0.0-alpha` release out.

**Comment by WRidder** [comment]
I can confirm that `3.0.0` espree fixes the issue. Tested this by manually installing the alpha in the eslint node_modules directory.

Thanks! I'll await the eslint update using espree 3.0.0 :)
